CHANGELOG.md

```diff
@@ -0,0 +1,246 @@
+# Change Log for AWS IoT Embedded C SDK
+
+## 202009.00 (September 14, 2020)
```

Comment:
when do we plan to merge this PR? At the day of release because of this info?

Reply:
I plan to re-target this PR to the `release-candidate` branch instead of waiting to merge this on the release date

Reply:
Change log should also go into master.

Reply:
Based on the new workflow being discussed, I will merge it to master only (and it will be carry-forwarded to RC)

Reply:
So if thats he case, I think we need to split this into 2 PRs. Before 9/14, master should get changelog upto v3x release. RC should have 202009.00 changelog. And then RC merges back to master and master gets it. Right?

Reply:
If the `release-candidate` branch is going to be merged back to `master` branch, then this PR can just be re-targeted to the `release-candidate` branch. 
@leegeth, what do you suggest?

Reply:
I think we can merge it to release-candidate as is. After the release is done, we may just need to update the CHANGELOG.md in master and we can create a PR to do that. We may not need to merge the release-candidate back to master at all though with the new flow. What do you all think about that?

Reply:
That works for me as long as we have a plan to port back changes from `release-candidate` to `master`